src/main/java/uk/gov/hmcts/ccd/domain/service/CreateOrUpdateUserProfileOperation.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.ccd.domain.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.jurisdiction.JurisdictionEntity;
+import uk.gov.hmcts.ccd.data.jurisdiction.JurisdictionRepository;
+import uk.gov.hmcts.ccd.data.userprofile.UserProfileRepository;
+import uk.gov.hmcts.ccd.domain.model.Jurisdiction;
+import uk.gov.hmcts.ccd.domain.model.UserProfile;
+import uk.gov.hmcts.ccd.endpoint.exception.BadRequestException;
+
+@Service
+public class CreateOrUpdateUserProfileOperation {
+
```

Comment:
What do you think SaveUserProfileOperation as an alternative to CreateOrUpdateUserProfileOperation

Reply:
201 for create and 204 for update

Reply:
First suggestion - adopted.
Second suggestion - here is @vlaurin's response on returning different HTTP success statuses from the same endpoint:
> Technically, you can, but it requires returning an instance of `Response<T>` instead of `T` directly, and then you have to build the `Response<T>` object manually
> creates a bit of overhead
> from a REST philosophy perspective, I don't know if it's frowned upon or not

So, I think I'll leave as is, i.e. return a `200` on save (whether it be a create or update)